ducktape/tests/reporter.py

```diff
@@ -63,7 +63,7 @@ def result_string(self):
             result_lines.append("    " + self.result.summary)
 
         if self.result.data is not None:
-            result_lines.append(json.dumps(self.result.data))
+            result_lines.append(pprint.pformat(self.result.data))
```

Comment:
Yeah, this was a bit iffy.
I wanted to support the use case where functions are used to parameterize tests, e.g. `@parameterize(x=my_function)`, and `json.dumps` doesn't know how to stringify functions.

This is also tricky if we also want to be able to specify which parameters to use in a test at the command-line.

rolling this back